website/content/docs/concepts/cel.mdx

```diff
@@ -0,0 +1,133 @@
+---
+sidebar_label: Common Expression Language in OpenBao
+description: How is Common Expression Language implemented in OpenBao.
+---
+
+# CEL in OpenBao
+
+The Common Expression Language (CEL) is used in PKI and JWT to declare validation rules
+and policies:
+- [CEL for JWT Auth](https://openbao.org/docs/rfcs/cel-jwt/)
+- [CEL in PKI](https://openbao.org/docs/rfcs/cel-pki/)
+
+The main CEL `Expression` in a `Celprogram` is typically one line of code. This can be broken down
+into multiple CEL expressions by using `CelVariable`'s. The following sections cover how to build
+a `CelProgram` in any part of OpenBao.
+
+---
+
+## Table of contents
+ - [Main expression](#main-expression)
+ - [Variables in a CelProgram](#variables-in-a-celprogram)
+ - [CelProgram](#celprogram)
+
+---
+
+## Main expression
+- This is the main expression that determines whether a CEL program has accepted or rejected
+a request.
+
+- The main expression in the `CelProgram` should return the engine's
+  specific output object on success, for instance:
+    * PKI → `ValidationOutput`
+    * JWT/OIDC → `pb.Auth`
```

Comment:
Should we move `ValidationOutput` into the `pb` package so these can be more similar?

Reply:
perhaps this should also have name like `PKIValidation`?

Reply:
Also, fwiw, since this shipped in v2.3.0, I wouldn't necessarily affect typing now if we can avoid it.